### PR TITLE
fix jshint errors

### PR DESCRIPTION
**Pull request by hthetiot**
Before:

> adhoc.js: line 2, col 5, Redefinition of 'URL'.
bootstrap.js: line 195, col 30, Expected '{' and instead saw 'loadCallback'.
browser.js: line 81, col 31, Missing '()' invoking a constructor.
browser.js: line 88, col 9, Function declarations should not be placed in blocks. Use a function expression or move the statement to the top of the outer function.
browser.js: line 104, col 2, Unnecessary semicolon.
require.js: line 56, col 12, Missing semicolon.
require.js: line 71, col 10, Missing semicolon.
require.js: line 123, col 35, Missing '()' invoking a constructor.
require.js: line 197, col 21, Comma warnings can be turned off with 'laxcomma'.
require.js: line 196, col 44, Misleading line break before ','; readers may interpret this as an expression boundary.
require.js: line 197, col 23, Misleading line break before ','; readers may interpret this as an expression boundary.
require.js: line 198, col 23, Misleading line break before ','; readers may interpret this as an expression boundary.
require.js: line 199, col 23, Misleading line break before ','; readers may interpret this as an expression boundary.
require.js: line 200, col 22, Misleading line break before ','; readers may interpret this as an expression boundary.
require.js: line 212, col 33, Misleading line break before '?'; readers may interpret this as an expression boundary.
require.js: line 213, col 55, Expected an assignment or function call and instead saw an expression.
require.js: line 219, col 25, Misleading line break before '?'; readers may interpret this as an expression boundary.
require.js: line 313, col 32, Missing '()' invoking a constructor.
require.js: line 520, col 65, Missing '()' invoking a constructor.
require.js: line 561, col 104, Missing semicolon.
require.js: line 569, col 95, Missing semicolon.
require.js: line 720, col 17, The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.
require.js: line 746, col 13, 'redirects' is already defined.
require.js: line 748, col 58, 'name' is already defined.
require.js: line 765, col 53, 'name' is already defined.
require.js: line 781, col 25, Missing '()' invoking a constructor.
require.js: line 782, col 37, Missing '()' invoking a constructor.
require.js: line 799, col 40, Expected '{' and instead saw 'return'.
require.js: line 800, col 80, Missing '()' invoking a constructor.
require.js: line 817, col 42, Missing '()' invoking a constructor.
require.js: line 1042, col 13, Expected an assignment or function call and instead saw an expression.
require.js: line 1061, col 41, Misleading line break before '?'; readers may interpret this as an expression boundary.
require.js: line 1066, col 29, Missing '()' invoking a constructor.
require.js: line 1070, col 30, Missing '()' invoking a constructor.
34 errors

After:

> browser.js: line 88, col 9, Function declarations should not be placed in blocks. Use a function expression or move the statement to the top of the outer function.
require.js: line 208, col 89, Expected an assignment or function call and instead saw an expression.
require.js: line 715, col 17, The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.
require.js: line 1041, col 13, Expected an assignment or function call and instead saw an expression.
4 errors`

### CODE REVIEW

File: bootstrap.js [hunk 1]
```diff
@@ -192,7 +192,9 @@
         var script = document.createElement("script");
         script.src = location;
         script.onload = function () {
-            if(loadCallback) loadCallback(script);
+            if(loadCallback) {
```
Comment (kriskowal) [inline bootstrap.js]: no rule for space between keyword and open paren?

File: require.js [hunk 2]
```diff
@@ -208,16 +204,14 @@
...
+                                (promises = [promises,iPromise])) : (promises = iPromise);
                         }
         			}
                 }
 
-                return promises
-                        ? (promises.push === void 0 ? promises : Promise.all(promises))
-                        : null;
+                return promises ? (promises.push === void 0 ? promises : 
+                            Promise.all(promises)) : null;
```
Comment (kriskowal) [inline require.js]: Does this improve readability?

File: require.js [hunk 3]
```diff
@@ -1031,16 +1028,18 @@
...
-                extension !== "js" &&
-                extension !== "json" &&
-                config.moduleTypes.indexOf(extension) === -1
-            )) {
+            if (
+                !extension || (
+                    extension !== "js" &&
+                        extension !== "json" &&
+                            config.moduleTypes.indexOf(extension) === -1
+                    )
```
Comment (kriskowal) [inline require.js]: I would suggest:

```
if (
    !extension || (
        extension != "js" &&
        extension != "json" &&
        config.modulesTypes.indexOf(extension) === -1
    )
) {
}
```

…to keep associative terms on the same indentation level and per K&R style match the close bracket to the indent level of the line that introduced the open bracket.

File: bootstrap.js (hunk 1 above)
Comment (hthetiot) [inline bootstrap.js]: Not jshint did not complain. I will do apply this keyword space rule in the future.

Comment (hthetiot) [inline bootstrap.js]: travis already perform 'jshint .'

File: require.js (hunk 3 above)
Comment (hthetiot) [inline require.js]: fixed

File: require.js (hunk 2 above)
Comment (hthetiot) [inline require.js]: jshint was complaining: Misleading line break before '?'; readers may interpret this as an expression boundary.

File: bootstrap.js (hunk 1 above)
Comment (hthetiot) [inline bootstrap.js]: done

**Comment by hthetiot** [comment]
@kriskowal I think I'm done here.
Fixed tests, merged jshint in test to pass travis build 'npm test' step.

**Pull request merged**